src/flownet/ahm/_run_ahm.py

```diff
@@ -215,6 +216,99 @@ def _get_distribution(
     return df
 
 
+def _constrain_using_well_logs(
+    porv_poro_trans_dist_values: pd.DataFrame,
+    data: np.ndarray,
+    network: NetworkModel,
+    measurement_type: str,
+    config: ConfigSuite.snapshot,
+) -> pd.DataFrame:
+    """
+    Function to constrain permeability and porosity distributions of flow tubes by using 3D kriging of
+    porosity and permeability values from well logs.
+
+    Args:
+        porv_poro_trans_dist_values: pre-constraining dataframe
+        data: well log data
+        network: FlowNet network model
+        measurement_type: 'prorosity' or 'permeability' (always log10)
+        config: FlowNet configparser snapshot
+
+    Returns:
+        Well-log constrained "porv_poro_trans_dist_values" DataFrame
+
+    """
+    n = config.flownet.constraining.kriging.n
+    n_lags = config.flownet.constraining.kriging.n_lags
+    anisotropy_scaling_z = config.flownet.constraining.kriging.anisotropy_scaling_z
+    variogram_model = config.flownet.constraining.kriging.variogram_model
+
+    if measurement_type == "permeability":
+        data[:, 3] = np.log10(data[:, 3])
+
+    variogram_parameters: Optional[Dict] = None
+    if measurement_type == "permeability":
+        variogram_parameters = dict(
+            config.flownet.constraining.kriging.permeability_variogram_parameters
+        )
+    elif measurement_type == "porosity":
+        variogram_parameters = dict(
+            config.flownet.constraining.kriging.porosity_variogram_parameters
+        )
+
+    if not variogram_parameters:
+        variogram_parameters = None
```

Comment:
When is this executing in this context? If `varioprogram_parameters` is an empty `{}`?

Reply:
Yes, so when the measurement type is set, but the variogram parameters are not set (that is, an empty dict - this is what configsuite spits out) then I actually want the None to be passed around and not an empty dict..